dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -406,9 +406,10 @@ internal string ExpandFromVariables(ContextVariables variables, string input)
 
```

Comment:
Why `OrderByDescending(m => m.Length)`?

Reply:
This ensures that longer variable names are replaced first before shorter variable names.